wordpress-mobile/WordPress-iOS
# Fix Google 2FA login

Fixes #13004 
WPAuth PR: https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/157

This uses the WPAuth changes to fix the Google 2FA login flow.

To test:

---
The flow that is fixed:
- Select Log In > Continue with Google.
- Select a Google account that is already connected to WordPress, and has 2FA enabled.
- Verify you get prompted for a verification code, and can log in.

<img width="400" alt="verification_prompt" src="https://user-images.githubusercontent.com/1816888/69590879-2fdd9b80-0fae-11ea-9a56-f7f5ed5b8f5c.png">

---
Since a refactor was needed, sanity check these cases to be sure I didn't break anything:

- Select a Google account that is already connected to WordPress, but does _not_ have 2FA enabled. You should get logged in.
- Select a Google account that matches an existing WP account email, but is not yet connected to a WP account. You should get prompted to connect them.
- Select a Google account that does not match a WP account email. You should see this list of options.

<img width="400" alt="what_do_i_do_now" src="https://user-images.githubusercontent.com/1816888/69591128-f9ece700-0fae-11ea-87c8-ca7499a62885.png">

---
PR submission checklist:
